plugins/auth/index.js

```diff
@@ -114,14 +114,29 @@ exports.register = (server, options, next) => {
                 allowCookieToken: false,
                 allowQueryToken: true,
                 validateFunc: (token, cb) => {
-                    if (token !== pluginOptions.temporaryAccessKey) {
-                        return cb(null, false, {});
+                    if (token === pluginOptions.temporaryAccessKey) {
+                        return cb(null, true, {
+                            username: pluginOptions.temporaryAccessUser,
+                            scope: ['user']
+                        });
                     }
 
-                    return cb(null, true, {
-                        username: pluginOptions.temporaryAccessUser,
-                        scope: ['user']
-                    });
+                    // Token is an API token
+                    // using function syntax makes 'this' the request
+                    // TODO: Should log that we're authenticating a user with a token
+                    const factory = server.app.userFactory;
```

Comment:
Eslint actually complains about it too, won't let you run npm test if you try to access `server._parent`. I think moving back to `function` notation and using the request is probably the correct choice.